lib/Doctrine/DBAL/Exception/FormatVariable.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Exception;
+
+use InvalidArgumentException;
+use function get_class;
+use function get_resource_type;
+use function gettype;
+use function is_array;
+use function is_bool;
+use function is_object;
+use function is_resource;
+use function is_string;
+use function sprintf;
+
+final class FormatVariable
+{
+    /**
+     * @param mixed $value
+     */
+    public function __invoke($value) : string
+    {
+        if ($value === null) {
+            return 'NULL';
+        }
+
+        if (is_object($value)) {
+            return get_class($value);
+        }
+
+        if (is_resource($value)) {
+            return get_resource_type($value);
+        }
+
+        if (is_bool($value)) {
+            return $value === true ? 'true' : 'false';
+        }
+
+        if (is_array($value)) {
+            return 'array';
+        }
+
+        if (is_string($value)) {
+            return $value;
```

Comment:
This probably should be just `'string'` since otherwise, it may display a long or security-sensitive string. With that, the class should be named as `GetVariableType`.